src/playground/App.js

```diff
@@ -169,6 +170,9 @@ const App = () => {
 		...options,
 		languageOptions: {
 			...options.languageOptions,
+			...(options?.languageOptions.parser && {
+				parser: typeScriptESLintParser,
+			}),
```

Comment:
We need to make sure to pass the `languageOptions.sourceType` correctly to `parserOptions.sourceType` in case of typeScriptESLintParser, More information here  https://github.com/eslint/eslint/pull/19563#discussion_r2018741729

Reply:
Done!